### PR TITLE
CPLAT-11606 Scan for Tool Directory

### PR DESCRIPTION
**Pull request by joebingham-wk**
## Motivation

The extension should be able to scan for the tool directory for cases when the directory is not in the content root.

## Changes
- Add a function that iterates up the directories looking for a tool directory until it reaches the content root.
- Update readme wording. We were updated by IT Ops that a ticket has been made to publish the extension in the marketplace, but the readme indicated that the extension is in the VS Code marketplace right now. This is now fixed!

## QA
- [x] ~Run tests. This can be done either by using VS Code's debugger interface or by running `yarn compile && yarn test` locally~ (crossing this one out because no unit tested code paths changed - just ones that need functional QA'ing)
- [ ] Verify that the extension works in the following environments using the .vsix file installation method (with the .vsix in this PR). "Extension works" means that the extension will correctly identify a tool directory nested within the project or fallback to the content root if none is found

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: src/extension.ts [hunk 1]
```diff
@@ -108,6 +108,23 @@ class RunFormatOnSave {
 		return process.execFile(executable, args, {cwd: this.projectDir});
 	}
 
+	getProjectPath(contentRoot : string, fileName : string) : string {
```
Comment (joebingham-wk) [inline src/extension.ts:111]: I did this pretty quickly to just get a POC out - in reality we'd probably want to use Node's [path parse API](https://nodejs.org/api/path.html#path_path_parse_path).

**Comment by smaifullerton-wk** [comment]
Is the lodash vulnerable dep a new issue @joebingham-wk ?

**Comment by joebingham-wk** [comment]
Talked about it offline, but yep, the lodash dep issue is new but there's a dependabot PR (#9)  to update that. RM had to update some settings around dependabot for #9 to merge though, and those will take around an hour to propagate. So when those should be out, I'll merge that pr and pull the changes into this one to make Aviary happy!

**Comment by joebingham-wk** [comment]
That dependency update is merged and Aviary is happy!

**Comment by evanweible-wf** [comment]
QA +1
- [x] Runs `dartfmt` in a project without an `over_react_format` dependency (or too old of a version)
- [x] Runs `over_react_format` in a project with an `over_react_format` dependency
- [x] Uses the config from the nearest root with a `tool/` dir if there are multiple in the open project

**Comment by joebingham-wk** [comment]
@Workiva/release-management-p

**Pull request merged**